operator/src/main/java/oracle/kubernetes/operator/helpers/PodDisruptionBudgetHelper.java

```diff
@@ -0,0 +1,328 @@
+
+      CreateResponseStep(String messageKey, Step next) {
+        super(next);
+        this.messageKey = messageKey;
+      }
+
+      @Override
+      public NextAction onFailure(Packet packet, CallResponse<V1beta1PodDisruptionBudget> callResponse) {
+        if (UnrecoverableErrorBuilder.isAsyncCallFailure(callResponse)) {
+          return updateDomainStatus(packet, callResponse);
+        } else {
+          return onFailure(getConflictStep(), packet, callResponse);
+        }
+      }
+
+      private NextAction updateDomainStatus(Packet packet, CallResponse<V1beta1PodDisruptionBudget> callResponse) {
+        return doNext(DomainStatusUpdater.createFailureRelatedSteps(callResponse, null), packet);
+      }
+
+      @Override
+      public NextAction onSuccess(Packet packet, CallResponse<V1beta1PodDisruptionBudget> callResponse) {
+        logPodDisruptionBudgetCreated(messageKey);
+        addPodDisruptionBudgetToRecord(callResponse.getResult());
+        return doNext(packet);
+      }
+    }
+
+    private class ReadResponseStep extends DefaultResponseStep<V1beta1PodDisruptionBudget> {
+      ReadResponseStep(Step next) {
+        super(next);
+      }
+
+      @Override
+      public NextAction onFailure(Packet packet, CallResponse<V1beta1PodDisruptionBudget> callResponse) {
+        return callResponse.getStatusCode() == CallBuilder.NOT_FOUND
+                ? onSuccess(packet, callResponse)
+                : onFailure(getConflictStep(), packet, callResponse);
+      }
+
+      @Override
+      public NextAction onSuccess(Packet packet, CallResponse<V1beta1PodDisruptionBudget> callResponse) {
+        V1beta1PodDisruptionBudget podDisruptionBudget = callResponse.getResult();
+        if (podDisruptionBudget == null) {
+          removePodDisruptionBudgetFromRecord();
+        } else {
+          addPodDisruptionBudgetToRecord(callResponse.getResult());
+        }
+        return doNext(packet);
+      }
+    }
+
+    private class ConflictStep extends Step {
+      @Override
+      public NextAction apply(Packet packet) {
+        return doNext(
+                new CallBuilder().readPodDisruptionBudgetAsync(getPDBName(), info.getNamespace(),
+                        new PodDisruptionBudgetContext.ReadResponseStep(conflictStep)), packet);
+      }
+
+      @Override
+      public boolean equals(Object other) {
+        if (other == this) {
+          return true;
+        }
+        if (!(other instanceof PodDisruptionBudgetHelper.PodDisruptionBudgetContext.ConflictStep)) {
+          return false;
+        }
+        PodDisruptionBudgetHelper.PodDisruptionBudgetContext.ConflictStep rhs =
+                ((PodDisruptionBudgetHelper.PodDisruptionBudgetContext.ConflictStep) other);
+        return new EqualsBuilder().append(conflictStep, rhs.getConflictStep()).isEquals();
+      }
+
+      private Step getConflictStep() {
+        return conflictStep;
+      }
+    }
+
+    Step verifyPodDisruptionBudget(Step next) {
```

Comment:
Do we need to update the domain status with Processing condition when there is work to be done to create or patch PDB?

Reply:
I was following the pattern in the ServiceHelper which is used to create server and cluster services. It doesn't seem to update the domain status with Progressing condition when creating services. Additionally, I think PDB is not as "important" to the domain status as services.

Reply:
It seems that currently we only update the status with Processing condition for pod/job creation and patching. We are probably okay without updating the status here.